connector-agent-python/src/lib.rs

```diff
@@ -72,3 +72,40 @@ fn read_pg(py: Python, conn: &str, sqls: Vec<String>, schema: &str) -> PyResult<
         .collect();
     PyResult::Ok(ret.into_py_dict(py).to_object(py))
 }
+
+#[pyfunction]
+fn read_sql<'a>(
+    py: Python<'a>,
+    conn: &str,
+    query: &str,
+    col: &str,
+    min: i64,
+    max: i64,
+    num: i64,
+    return_type: &str,
+) -> PyResult<&'a PyAny> {
+    let mut queries: Vec<String> = vec![];
+    let partition_size = match (max - min + 1) % num == 0 {
+        true => (max - min + 1) / num,
+        false => (max - min + 1) / num + 1,
+    };
+
+    for i in 0..num {
+        let lower = min + i * partition_size;
+        let upper = min + (i + 1) * partition_size;
+        let partition_query = format!(
```

Comment:
Input queries may have predicates. For example, if we have a user input query: `SELECT * FROM r WHERE a > 10`, then the partition queries will become: `SELECT * FROM r WHERE a > 10 WHERE col >= lower AND  col < upper`. Maybe you can use `sqlparse` to parse the query first and then modify the query (like in `connector-agent/src/data_sources/postgres/sql.rs`) by adding additional selection

Reply:
We'll look into this.

Reply:
Yup, it makes sense~

Reply:
Actually just tried to test this query with some values: SELECT * FROM r WHERE a > 10

This is the output: ["SELECT * FROM r WHERE a > 10 where id >= 5 and id < 9", "SELECT * FROM r WHERE a > 10 where id >= 9 and id < 13", "SELECT * FROM r WHERE a > 10 where id >= 13 and id < 17"]

So I think it might work? 🤔